Show warnings when sync failed after External Service creation/update was successful (#4148)
sourcegraph/sourcegraph: CHANGELOG.md

### Changed

- The saved searches UI has changed. There is now a Saved searches page in the user and organizations settings area. A saved search appears in the settings area of the user or organization it is associated with.

> - Site admins will now see a warning if creating or updating an external service was successful but the process could not complete entirely due to an ephemeral error (such as the GitHub API shedding load and returning incomplete results).

Reviewer: ```suggestion
- Site admins will now see a warning if creating or updating an external service was successful but the process could not complete entirely due to an ephemeral error (such as when the code host's API is unreachable due to an outage).
```

I'm not sure what "shedding load" means in this case, so I reworded it to something that makes sense to me. Let me know if "shedding load" means something specific.

Author: Specifically, it's about this: https://developer.github.com/v3/search/#timeouts-and-incomplete-results and https://developer.github.com/changes/2014-04-07-understanding-search-results-and-potential-timeouts/

> For queries that exceed the time limit, the API returns the matches that were already found prior to the timeout, and the response has the incomplete_results property set to true.

My original changelog entry was... a guess, because I didn't know what to write, but I now think this would be better:

```
(such as GitHub API search queries running into timeouts and returning incomplete results)
```

Reviewer: 👍 looks good
